### PR TITLE
Safer cursor overriding for python

### PR DESCRIPTION
**Pull request by m-kuhn**
```
with OverrideCursor(Qt.WaitCursor):
    do_something_useful()
```

because it handles exceptions and intermediate `return` statements and it's shorter than writing `try` and `finally` each time.

### CODE REVIEW

File: python/plugins/processing/algs/qgis/ui/FieldsCalculatorDialog.py [hunk 1]
```diff
@@ -220,10 +221,9 @@ def getParamValues(self):
 
     def accept(self):
         keepOpen = ProcessingConfig.getSetting(ProcessingConfig.KEEP_DIALOG_OPEN)
-        try:
-            parameters = self.getParamValues()
-            if parameters:
-                QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
+        parameters = self.getParamValues()
+        if parameters:
+            with OverrideCusor(Qt.WaitCursor):
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/ui/FieldsCalculatorDialog.py]: "Cusor" -> Cursor

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Looks great - nice and clean!

**Comment by NathanW2** [comment]
Nice. Merge it

**Comment by nyalldawson** [comment]
Travis says no!

**Pull request merged**